tests/frame/with_row_index_test.py

```diff
@@ -12,5 +12,5 @@
 
 def test_with_row_index(constructor: Constructor) -> None:
     result = nw.from_native(constructor(data)).with_row_index()
-    expected = {"a": ["foo", "bars"], "ab": ["foo", "bars"], "index": [0, 1]}
+    expected = {"index": [0, 1], "a": ["foo", "bars"], "ab": ["foo", "bars"]}
```

Comment:
Order is not checked anyway

Reply:
can we make `assert_equal_data` check the order?

Reply:
Nice one! Turns out pyarrow was also wrong! Should be fixed now